2018/day-2/InventoryManagementSystem.hs

```diff
@@ -0,0 +1,32 @@
+-- https://adventofcode.com/2018/day/2
+
+module InventoryManagementSystem where
+
+import Data.List (group, sort, intersect, (\\))
+
+readLines :: String -> IO [String]
+readLines = fmap lines . readFile
+
+calculateChecksum :: [String] -> Int
+calculateChecksum list = countTwos * countThrees
+  where
+  countTwos = countOccurences 2 list
+  countThrees = countOccurences 3 list
+
+countOccurences :: Int -> [String] -> Int
+countOccurences value list = length (filter (any (== value)) (map toOccurences list))
+
+toOccurences :: String -> [Int]
+toOccurences = map length . group . sort
+
+commonLetters :: [String] -> String
+commonLetters list = head [intersect xs ys | xs <- list, ys <- list, differByOneCharacter xs ys]
+
+differByOneCharacter :: String -> String -> Bool
+differByOneCharacter xs ys = length (xs \\ ys) == 1
```

Comment:
I suspect this might be incorrect. `\\` ignores position and the exercise wants us to make sure strings are different by one letter _at the same position_:
>The boxes will have IDs which differ by exactly one character at the same position in both strings

Reply:
Yeah, you're right. At first I had a version with `filter`, which gave me a correct solution, and then wanted to make it more _idiomatic_ before pushing. I suppose I've been too confident 🙃

Reply:
Fix that:
https://github.com/walerian777/advent-of-code/blob/d5e08e4a466104f40ab523fe47b8274e6039bd44/2018/day-2/InventoryManagementSystem.hs#L25-L26